### PR TITLE
MAINT/DOC: clean up `_lib._array_api`, update docs

### PR DESCRIPTION
**Pull request by lucascolley**
#### Reference issue
Follow-up to gh-21294

#### What does this implement/fix?
- First commit improves the naming consistency in `_lib._array_api` and updates `__all__`.
- Second commit is a full sweep of updates on the docs page to accommodate the requested improvements in gh-21294.

### CODE REVIEW

File: doc/source/dev/api-dev/array_api.rst [hunk 1]
```diff
@@ -183,11 +176,13 @@ Input array validation uses the following pattern::
    # alternatively, if there are multiple array inputs, include them all:
    xp = array_namespace(arr1, arr2)
 
+   # replace np.asarray` with `xp.asarray
```
Comment (lucascolley) [inline doc/source/dev/api-dev/array_api.rst:179]:
```suggestion
   # replace np.asarray with xp.asarray
```

**Comment by rgommers** [comment]
Could you resolve the merge conflicts here Lucas?

**Comment by lucascolley** [comment]
> Could you resolve the merge conflicts here Lucas?

done!

**Review by rgommers — APPROVED** [review]
Thanks @lucascolley, this all looks good to me. Very easy to review, zero comments:)

**Pull request merged**